src/spawn.rs

```diff
@@ -7,18 +7,31 @@ use std::ops::{Deref, DerefMut};
 use std::path::Path;
 use std::process::{Command, Output, Stdio};
 
-use serde::Serialize;
+use serde::{Serialize, Serializer};
 
 #[derive(Serialize)]
 pub struct Info {
     program: String,
     args: Vec<String>,
     #[serde(skip_serializing_if = "HashMap::is_empty")]
+    #[serde(serialize_with = "ordered_map")]
     env: HashMap<String, String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     stdin: Option<String>,
 }
 
+/// For use with serde's [serialize_with] attribute
+fn ordered_map<S, K: Ord + Serialize, V: Serialize>(
+    value: &HashMap<K, V>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let ordered: BTreeMap<_, _> = value.iter().collect();
```

Comment:
Another option would be to use a `BTreeMap` for the original data structure; lmk your preferences